parser/src/fstring.rs

```diff
@@ -30,17 +30,39 @@ impl<'a> FStringParser<'a> {
         while let Some(ch) = self.chars.next() {
             match ch {
                 '!' if delims.is_empty() => {
-                    conversion = Some(match self.chars.next() {
-                        Some('s') => ConversionFlag::Str,
-                        Some('a') => ConversionFlag::Ascii,
-                        Some('r') => ConversionFlag::Repr,
-                        Some(_) => {
-                            return Err(InvalidConversionFlag);
+                    let x = self.chars.next();
+                    if let Some('=') = x {
+                        expression.push(ch);
+                        expression.push('=');
+                    } else {
+                        if expression.is_empty() {
+                            return Err(EmptyExpression);
                         }
-                        None => {
-                            break;
+
+                        conversion = Some(match x {
+                            Some('s') => ConversionFlag::Str,
+                            Some('a') => ConversionFlag::Ascii,
+                            Some('r') => ConversionFlag::Repr,
+                            Some(_) => {
+                                return Err(InvalidConversionFlag);
+                            }
+                            None => {
+                                break;
+                            }
```

Comment:
It seems like `None` after the conversion character still yields an invalid conversion flag error:
```py
>>> f"{a!}"
  File "<stdin>", line 1
SyntaxError: f-string: invalid conversion character: expected 's', 'r', or 'a'
```
You could do `Some(_) | None => {...}`

Reply:
This isn't an instance of the None case, which would be f"{a!", although I did add tests for both cases.